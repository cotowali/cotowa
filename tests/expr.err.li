fn f(a int) int {
  return a
}

echo(x)

f("a") + "a"
let x = "a" + f("a")

y = 10
y = "10"
0 = 10

f("a") |> f("a")

let arr = [0]
arr['10']
arr[0][0]
x[0]
