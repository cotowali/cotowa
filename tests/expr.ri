6 / 2 + 3 * 3
1 + 1 - 1
6 / 2 % 2
(1 + 2) * (4 - 2)
let v = 10
-v
+v

fn zero() int {
  let v = 0

  // evaluated but not used. this is valid but no output
  0
  v
  add(1, 2)

  return v
}

fn add(a int, b int) int {
  return a + b
}

fn as_int(v any) int {
  return v as int
}

zero()
add(3, 4)
add(as_int("1"), 2)
