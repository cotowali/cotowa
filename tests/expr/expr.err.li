fn f(a: int) int {
  return a
}

echo(x)

f("a") + "a"
var x = "a" + f("a")

y = 10
y = "10"
0 = 10

f("a") |> f("a")
0 |> 0

var arr = [0]
arr['10']
arr[0][0]
x[0]

fn int |> g() {
  read(&n)

  var s: string
  read(&s)
}
