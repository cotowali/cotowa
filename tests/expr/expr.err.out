error: expr.err.li 5,6: variable `x` is not defined
    5 | echo(x)
      |      ^
error: expr.err.li 7,1: mismatched types: `int` (left) and `string` (right)
    7 | f("a") + "a"
      | ^^^^^^^^^^^^
error: expr.err.li 7,3: mismatched types: `int` (expected) and `string` (found)
    7 | f("a") + "a"
      |   ^^^
error: expr.err.li 8,9: mismatched types: `string` (left) and `int` (right)
    8 | var x = "a" + f("a")
      |         ^^^^^^^^^^^^
error: expr.err.li 8,17: mismatched types: `int` (expected) and `string` (found)
    8 | var x = "a" + f("a")
      |                 ^^^
error: expr.err.li 10,1: variable `y` is not defined
   10 | y = 10
      | ^
error: expr.err.li 11,1: variable `y` is not defined
   11 | y = "10"
      | ^
error: expr.err.li 12,1: invalid left-hand side of assignment
   12 | 0 = 10
      | ^
error: expr.err.li 14,1: mismatched types: `int` (left) and `void` (pipe in of right)
   14 | f("a") |> f("a")
      | ^^^^^^^^^^^^^^^^
error: expr.err.li 14,3: mismatched types: `int` (expected) and `string` (found)
   14 | f("a") |> f("a")
      |   ^^^
error: expr.err.li 14,13: mismatched types: `int` (expected) and `string` (found)
   14 | f("a") |> f("a")
      |             ^^^
error: expr.err.li 15,6: expected function call, but found int
   15 | 0 |> 0
      |      ^
error: expr.err.li 18,5: mismatched types: `int` (expected) and `string` (found)
   18 | arr['10']
      |     ^^^^
error: expr.err.li 19,1: `int` does not support indexing
   19 | arr[0][0]
      | ^^^^^^^^^
error: expr.err.li 20,1: `int` does not support indexing
   20 | x[0]
      | ^^^^
error: expr.err.li 21,1: variable `foo` is not defined
   21 | foo[0]
      | ^^^
error: expr.err.li 24,9: variable `n` is not defined
   24 |   read(&n)
      |         ^
error: expr.err.li 27,8: mismatched types: `&int` (expected) and `&string` (found)
   27 |   read(&s)
      |        ^^
error: expr.err.li 30,1: invalid operation: `string` ** `int`
   30 | "2" ** 2
      | ^^^^^^^^
error: expr.err.li 31,1: invalid operation: `int` ** `string`
   31 | 2 ** "2"
      | ^^^^^^^^
